BuildResidentialHPXML/measure.rb

```diff
@@ -6317,10 +6317,13 @@ def self.set_water_heating_systems(hpxml_bldg, args)
       end
     end
 
-    if args[:water_heater_num_units_served] > 1
-      is_shared_system = true
-      number_of_units_served = args[:water_heater_num_units_served]
+    if [HPXML::ResidentialTypeSFA, HPXML::ResidentialTypeApartment].include? args[:geometry_unit_type]
+      if args[:water_heater_num_bedrooms_served].get > args[:geometry_unit_num_bedrooms]
```

Comment:
So if they don't provide this input, they get a cryptic `.get` error message, right?

How about this instead:
```suggestion
      if args[:water_heater_num_bedrooms_served].to_f > args[:geometry_unit_num_bedrooms]
```

And then change the argument description from "Required if #{HPXML::ResidentialTypeSFA} or #{HPXML::ResidentialTypeApartment}" to "Only needed if #{HPXML::ResidentialTypeSFA} or #{HPXML::ResidentialTypeApartment} and it is a shared water heater serving multiple dwelling units."

That way they can simply leave the input blank if it's not a shared system.

Reply:
Make the same changes for PV/batteries, as they also have `foo_num_bedrooms_served` inputs.